partner_credit_limit/migrations/11.0.0.0.2/post-migration.py

```diff
@@ -0,0 +1,28 @@
+from odoo import api, SUPERUSER_ID
+
+
+def migrate(cr, version):
+    set_credit_limit_values_to_partner(cr)
+
+
+def set_credit_limit_values_to_partner(cr):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    env.cr.execute(
+        """
+        SELECT id, credit_limit, grace_payment_days FROM res_partner
+        WHERE credit_limit != 0 OR grace_payment_days != 0""")
+    data = env.cr.dictfetchall()
+    credit_limit_vals = {
+        p['id']: {
+            'credit_limit': p['credit_limit'],
+            'grace_payment_days': p['grace_payment_days']} for p in data}
+    partner_ids = [p.get('id') for p in data]
+    partners = env['res.partner'].browse(partner_ids or [])
+    companies = env['res.company'].search([])
+    for company in companies:
+        for partner in partners:
+            partner.with_context(force_company=company.id).write({
+                'credit_limit': credit_limit_vals.get(partner.id, {}).get(
+                    'credit_limit', 0),
+                'grace_payment_days': credit_limit_vals.get(
+                    partner.id, {}).get('grace_payment_days', 0)})
```

Comment:
`ALTER TABLE res_partner DROP COLUMN credit_limit;`
`ALTER TABLE res_partner DROP COLUMN grace_payment_days;`

Reply:
Why remove it?
I think we should keep it in order to have that values for the record.

Regards!

Reply:
Better using [ir_model._drop_column method](https://github.com/odoo/odoo/blob/9d7147a3c189d9e2edcdeb0e5cf28b75cf7f366b/odoo/addons/base/ir/ir_model.py#L524)

Because it shouldn't used anymore.
If you have a script using this value directly from database so it will use a deprecated field and outdated value silently.

We need a raise or real fail in other to fix it.

But if you need storing for the record (good point here), you can use another way (Storing a `ir_log` with `data` value)

Reply:
Got it